build/gtk-bundle-from-msys2.sh

```diff
@@ -0,0 +1,259 @@
+glib-networking
+gmp
+gnupg
+gnutls
+gpgme
+gstreamer
+gst-plugins-base
+gtkspell
+http-parser
+hunspell
+icu
+libassuan
+libgcrypt
+libgpg-error
+libgit2
+libidn
+libjpeg-turbo
+libogg
+libsoup
+libssh2
+libsystre
+libtasn1
+libtheora
+libtiff
+libtre-git
+libvorbis
+libvorbisidec-svn
+libwebp
+libxml2
+libxslt
+lua51
+nettle
+openssl
+orc
+p11-kit
+readline
+rtmpdump-git
+sqlite3
+termcap
+"
+
+gtk2_dependency_pkgs="
+webkitgtk2
+"
+gtk3_dependency_pkgs="
+webkitgtk3
+"
+
+
+handle_command_line_options() {
+	for opt in "$@"; do
+		case "$opt" in
+		"-c"|"--cache")
+			use_cache="yes"
+			;;
+		"-z"|"--zip")
+			make_zip="yes"
+			;;
+		"-2")
+			gtkv="2"
+			;;
+		"-3")
+			gtkv="3"
+			;;
+		"-h"|"--help")
+			echo "gtk-bundle-from-msys2.sh [-c] [-z] [-2 | -3]"
+			echo "   -c Use pacman cache. Otherwise pacman will download"
+			echo "      archive files"
+			echo "   -z Create a zip afterwards"
+			echo "   -2 Prefer gtk2"
+			echo "   -3 Prefer gtk3"
+			exit 1
+			;;
+		*)
+			cachedir="$opt"
+			;;
+		esac
+	done
+}
+
+initialize() {
+	if [ -z "$cachedir" ]; then
+		cachedir="/var/cache/pacman/pkg"
+	fi
+
+	if [ "$use_cache" = "yes" ] && ! [ -d "$cachedir" ]; then
+		echo "Cache dir \"$cachedir\" not a directory"
+		exit 1
+	fi
+
+	gtk="gtk$gtkv"
+	eval "gtk_dependency_pkgs=\${${gtk}_dependency_pkgs}"
+
+	pkgs="
+${packages}
+${gtk_dependency_pkgs}
+"
+}
+
+_getpkg() {
+	if [ "$use_cache" = "yes" ]; then
+		package_info=`pacman -Qi mingw-w64-$ABI-$1`
+		package_version=`echo "$package_info" | grep "^Version " | cut -d':' -f 2 | tr -d '[[:space:]]'`
+		ls $cachedir/mingw-w64-${ABI}-${1}-${package_version}-* | sort -V | tail -n 1
+	else
+		pacman -Sp mingw-w64-${ABI}-${1}
+	fi
+}
+
+extract_packages() {
+	# hack for libxml2 postinstall script which expects "bin/mkdir"
+	mkdir -p bin
+	cp /bin/mkdir bin/
+	# extract packages
+	for i in $pkgs; do
+		pkg=$(_getpkg $i)
+		if [ "$use_cache" = "yes" ]; then
+			if [ -e "$pkg" ]; then
+				echo "Extracting $pkg from cache"
+				tar xaf $pkg
+			else
+				echo "ERROR: File $pkg not found"
+				exit 1
+			fi
+		else
+			echo "Download $pkg using curl"
+			curl -L "$pkg" | tar -x --xz
+		fi
+		if [ -f .INSTALL ]; then
+			echo "Running post_install script for \"$i\""
+			/bin/bash -c ". .INSTALL; post_install"
+		fi
+		rm -f .INSTALL .MTREE .PKGINFO .BUILDINFO
+	done
+}
+
+move_extracted_files() {
+	if [ -d mingw32 ]; then
+		for d in bin etc home include lib libexec locale sbin share ssl var; do
+			if [ -d "mingw32/$d" ]; then
+				rm -rf $d
+				# prevent sporadic 'permission denied' errors on my system, not sure why they happen
+				sleep 0.5
+				mv mingw32/$d .
+			fi
+		done
+		rmdir mingw32
+	fi
+}
+
+cleanup_unnecessary_files() {
+	echo "Cleanup unnecessary files"
+	# etc: cleanup development files
+	rm -rf etc/bash_completion.d
+	rm -rf etc/dbus-1
+	rm -rf etc/pkcs11
+	rm -rf etc/xml
+	# include: cleanup development files
+	rm -rf include
+	# lib: cleanup development files
+	rm -rf libexec
```

Comment:
Yeah, the comment is a bit misleading. Sorry.
```
$ ls -l bundle-plugins/libexec/*
-rwxr-xr-x 1 enrico None  71586 Jul 10  2015 bundle-plugins/libexec/dbus-bash-completion-helper.exe
-rwxr-xr-x 1 enrico None  22546 Jul 10  2015 bundle-plugins/libexec/geoclue-example.exe
-rwxr-xr-x 1 enrico None  26689 Jul 10  2015 bundle-plugins/libexec/geoclue-geonames.exe
-rwxr-xr-x 1 enrico None  54614 Jul 10  2015 bundle-plugins/libexec/geoclue-gsmloc.exe
-rwxr-xr-x 1 enrico None  64544 Jul 10  2015 bundle-plugins/libexec/geoclue-hostip.exe
-rwxr-xr-x 1 enrico None  81757 Jul 10  2015 bundle-plugins/libexec/geoclue-localnet.exe
-rwxr-xr-x 1 enrico None  27452 Jul 10  2015 bundle-plugins/libexec/geoclue-manual.exe
-rwxr-xr-x 1 enrico None 106206 Jul 10  2015 bundle-plugins/libexec/geoclue-master.exe
-rwxr-xr-x 1 enrico None  27775 Jul 10  2015 bundle-plugins/libexec/geoclue-nominatim.exe
-rwxr-xr-x 1 enrico None  78317 Jul 10  2015 bundle-plugins/libexec/geoclue-plazes.exe
-rwxr-xr-x 1 enrico None  78238 Jul 10  2015 bundle-plugins/libexec/geoclue-skyhook.exe
-rwxr-xr-x 1 enrico None  24743 Jul 10  2015 bundle-plugins/libexec/geoclue-yahoo.exe

bundle-plugins/libexec/gstreamer-1.0:
total 20
-rwxr-xr-x 1 enrico None 20137 Feb 10 09:29 gst-plugin-scanner.exe
```
Nothing we need.
Anyway, I'll change the script to let `libexec` alone and instead delete all *.exe. Then the resulting empty directories will be deleted at the end.